### PR TITLE
fix(WEB-7151): Typescript check: Allow typed JS

### PR DESCRIPTION
**Pull request by georgegillams**
## Jira

[WEB-7151](https://typeform.atlassian.net/browse/WEB-7151)

## Backstory

Some files can't be renamed to have a `.ts` or `.tsx` extension. For example, `next.config.js` and any files that it requires.
https://nextjs.org/docs/pages/building-your-application/configuring/typescript#type-checking-nextconfigjs

However, we can still benefit from Typescript in these files by adding `// @ts-check` at the top of the file. eg https://github.com/Typeform/public-main-site/pull/4089

## The solution

This changes the typescript check to allow JS files with such an annotation. This will avoid false-negatives when Typescript is used this way, and will encourage Typeformers to adopt JS in these places to make adoption % look better 🙌

## Testing it out the wild

In this PR, the check passes (because the JS file I've edited is type-checked)
https://github.com/Typeform/public-main-site/pull/4091

And in this PR, I edit a regular JS file and so it complains
https://github.com/Typeform/public-main-site/pull/4095

[WEB-7151]: https://typeform.atlassian.net/browse/WEB-7151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/checks/requiredTypeScript.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+import ignore, { Ignore as IgnoredFileFilter } from 'ignore'
 
 import { github } from '../infrastructure/github'
 import * as fs from '../infrastructure/fs'
 import { isBot } from '../conditions/triggeredByBot'
 
 import Check from './check'
-import ignore, { Ignore as IgnoredFileFilter } from 'ignore'
+
+const JS_TS_CHECK_COMMENT = '// @ts-check'
```
Comment (trapped) [inline src/checks/requiredTypeScript.ts:12]: Asking out of ignorance: is the whitespace optional here? e.g. would `//@ts-check` be accepted by `tsc`?

File: src/checks/requiredTypeScript.ts [hunk 2]
```diff
@@ -59,12 +75,23 @@ async function checkPullRequest(): Promise<boolean> {
...
-  const jsFiles = files.filter((f) => isForbiddenJSFile(f.filename, filter))
-  const renamedJsFiles = files.filter(
+  const filesWithContent = await Promise.all(
+    files.map((f) =>
+      (async () => {
+        const fileContent = await getFileContent(f.filename)
+        return { ...f, fileContent }
+      })()
+    )
+  )
```
Comment (trapped) [inline src/checks/requiredTypeScript.ts:85]: Do the files need to be fetched or are they available in the filesystem already?

**Review by trapped — APPROVED** [review]
Left two questions but overall LGTM. Thanks for contributing to making all of Typeform's codebases better!

File: src/checks/requiredTypeScript.ts (hunk 1 above)
Comment (georgegillams) [reply src/checks/requiredTypeScript.ts:12]: Good catch. I'll change it to a regex that accepts any amount of whitespace 👍

File: src/checks/requiredTypeScript.ts (hunk 2 above)
Comment (georgegillams) [reply src/checks/requiredTypeScript.ts:85]: We could actually read them from the file-system 👍

File: src/checks/requiredTypeScript.ts (hunk 1 above)
Comment (trapped) [reply src/checks/requiredTypeScript.ts:12]: Great!

File: src/checks/requiredTypeScript.ts (hunk 2 above)
Comment (trapped) [reply src/checks/requiredTypeScript.ts]: Glad to hear that - it will definitely improve the performance of the check 🙂

**Review by pablonaranjo — APPROVED** [review]

**Pull request merged**